tools/nnyeah/nnyeah/Program.cs

```diff
@@ -7,8 +7,30 @@
 using System.Diagnostics.CodeAnalysis;
 
 namespace Microsoft.MaciOS.Nnyeah {
+	public class ConversionException : Exception {
+		public ConversionException (string message) : base (message)
+		{
+		}
+
+		public ConversionException (string message, params object? [] args) : base (string.Format(message, args))
+		{
+		}
+	}
+
 	public class Program {
-		static void Main (string [] args)
+		static int Main (string [] args)
+		{
+			try {
+				Main2 (args);
+				return 0;
+			}
+			catch (ConversionException c) {
+				Console.Error.WriteLine (c.Message);
+				return 1;
+			}
```

Comment:
You'll want to catch all exceptions too, otherwise the process will exit with a very ugly message.

Reply:
Alright, I wasn't sure here.

Reply:
Fixed

Reply:
I'd add the following, do catch our base exception, then catch all exceptions and do a console.writeline with the message and then the url to provide an issue. If we got an unexpected exception it should be a bug/issue just like we do in xamarin.